store/pogreb/pogreb.go

```diff
@@ -64,6 +71,12 @@ func (s *pStorage) Put(c cid.Cid, entry entry.Value) (bool, error) {
 }
 
 func (s *pStorage) put(k []byte, in entry.Value) (bool, error) {
+	// Acquire lock
+	err := s.lock(k)
+	if err != nil {
```

Comment:
I really do not like that lock acquisition can fail (even if only in theory).  Look at https://github.com/im7mortal/kmutex or https://pkg.go.dev/k8s.io/utils/keymutex instead.